.github/workflows/wait_for_required_workflows.yml

```diff
@@ -0,0 +1,73 @@
+name: Wait for all required workflows to pass
+
+on: [pull_request, push]
+
+jobs:
+  wait_for_required_workflows:
+    name: wait-for-required-workflows
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2 # OR "2" -> To retrieve the preceding commit.
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v31
+        with:
+          dir_names: true
+      - uses: actions/github-script@v6
+        env:
+          DIRS: ${{ steps.changed-files.outputs.all_changed_files }}
+          REF: ${{ github.ref }}
+        with:
+          script: |
+            const { REF, DIRS } = process.env
+            let now = new Date().getTime()
+            const deadline = now + 50 * 1000
```

Comment:
Does this mean we only allow 50 seconds for tests to run? We may want to increase that to ~10 minutes.

Reply:
Ah good catch. 15 min sounds good, anyway if a workflow fails earlier it will stop.